src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static killAnimal(animal) {
+    delete Animal.alive[Animal.alive.indexOf(animal)];
+  }
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (!prey.hidden && prey instanceof Herbivore) {
+      prey.health -= 50;
+    }
+
+    if (prey.health < 1) {
+      Animal.killAnimal(prey);
```

Comment:
You don't need here any other methods, let's just filter `Animal.alive` array.

[Checklist](https://github.com/mate-academy/js_herbivores_and_carnivores/blob/master/checklist.md) the 3rd point